### PR TITLE
Support for target-typed new in Insecure XSLT script processing (CA3076)

### LINKED ISSUE #6394: Make sure `DoNotUseInsecureXSLTScriptExecution` handles target-typed new
Also this one may have the same bug:

https://github.com/dotnet/roslyn-analyzers/blob/7c544ae66c44f1705ef583b112def4db52c0179a/src/NetAnalyzers/Core/Microsoft.NetFramework.Analyzers/DoNotUseInsecureXSLTScriptExecution.cs#L230

_Originally posted by @Youssef1313 in https://github.com/dotnet/roslyn-analyzers/pull/5831#pullrequestreview-868985275_

### PR DESCRIPTION
**Pull request by kandicst**
Fixes #6394.

### CODE REVIEW

File: src/NetAnalyzers/RulesMissingDocumentation.md [hunk 1]
```diff
@@ -9,5 +9,4 @@ CA1513 | <https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-
 CA1856 | <https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1856> | Incorrect usage of ConstantExpected attribute |
 CA1857 | <https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1857> | A constant is expected for the parameter |
 CA1859 | <https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1859> | Use concrete types when possible for improved performance |
-CA1860 | <https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1860> | Avoid using 'Enumerable.Any()' extension method |
```
Comment (drewnoakes) [inline src/NetAnalyzers/RulesMissingDocumentation.md:12]: Intentional? Seems unrelated to this PR.

Comment (kandicst) [reply src/NetAnalyzers/RulesMissingDocumentation.md]: This was autogenerated from when I ran the build. I removed it now.

**Review by mavasani — APPROVED** [review]
